### ISSUE TITLE
Pub/Sub socket stats: dropped messages due to full buffers

### ISSUE THREAD

**Issue by mempirate** [chainbound/msg-rs#37]
We need some stats for tracking the number of lost messages due to full buffers inside of `PubSocket` and `SubSocket`.

**Comment by PatStiles** [comment]
@mempirate down to take this on!

**Comment by mempirate** [comment]
I think I wouldn't want to have internal Prometheus metrics, the idea of the stats is that any consumer of this library can periodically call it to update their own metrics with whatever stack they're using! Would like this to be as lightweight as possible.

**Comment by mempirate** [comment]
What I had in mind was actually some metrics here (i.e. on the receiving side of the socket):

https://github.com/chainbound/msg-rs/blob/87d105ba824ae8b456dd7c665388f7fe685e492d/msg-socket/src/sub/driver.rs#L356-L364

I don't think we need stats on dropped sends if there are no active sessions.

**Comment by 7suyash7** [comment]
Hi @mempirate , is this available to be worked on? Happy to pick it up if so

**Comment by mempirate** [comment]
Yes, this is available to work on! Let me know if you have any questions.
